cmake/ConfigModuleDirs.cmake

```diff
@@ -0,0 +1,39 @@
+#Function which (a) sets the module output directory for a target, and (b) includes other module directories
+#preserves the include directories already set on the target, adding them at the end of the list.
+
+# It removes all previous include directories pointing to Amber's modules folder,
+# so that if you use copy_target() you can set new module paths without leaving the old ones lingering
+# at the end of the include path like a bad smell.
+
+function(config_module_dirs TARGETNAME TARGET_MODULE_DIR) #3rd optional argument: extra module include directories
+
```

Comment:
In the call `config_module_dirs(libquick quick/serial libxc/serial)`, `libquick` is ${TARGETNAME}, are both `quick/serial` and `libxc/serial` ${TARGET_MODULE_DIR} ? If so, how do CMAKE do the split?

Reply:
These are just file paths under the amber_modules directory, they don't reference target names at all.

Reply:
So what is the variable that `libquick` corresponds to in this call?

Reply:
I feel that this function is important if I want to build different versions of `QUICK` at the same time.

Reply:
libquick corresponds to TARGETNAME, yeah.  It's the first argument in the function call, so it goes to the first argument in the list.